### ISSUE TITLE
Generator fails on OSX

### ISSUE THREAD

**Issue by brokyo** [meanjs/generator-meanjs#133]
The Yo generator on master throws a bunch of errors that I've not seen mentioned in the other posts and I'm afraid I don't understand much about them. I can't attach the log and it's too big to screenshot but if there's something smarter I can do to share info please let me know.

My apologies if this has previous been covered but I'm out of my depth at this point and haven't been able to take anything from the other answers.

This particular error is on master with

$ node -v
v4.2.1.
$ npm -v
2.14.7
$ yo --version
1.4.8

while I'm using OSX 10.10.5

but I downgraded to the explicitly supported versions (node 0.10.36 and npm 1.3.8) and had the same result
```
make: *** [Release/obj.target/bufferutil/src/bufferutil.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/me/mean/test/node_modules/grunt-node-inspector/node_modules/node-inspector/node_modules/ws/node_modules/bufferutil
gyp ERR! node -v v4.2.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
```

```
make: *** [Release/obj.target/validation/src/validation.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/me/mean/test/node_modules/grunt-node-inspector/node_modules/node-inspector/node_modules/ws/node_modules/utf-8-validate
gyp ERR! node -v v4.2.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
```

```
make: *** [Release/obj.target/debug/src/debug.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:270:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Darwin 14.5.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/me/mean/test/node_modules/grunt-node-inspector/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.6/node-v46-darwin-x64/debug.node" "--module_name=debug" "--module_path=/Users/me/mean/test/node_modules/grunt-node-inspector/node_modules/node-inspector/node_modules/v8-debug/build/debug/v0.4.6/node-v46-darwin-x64"
gyp ERR! cwd /Users/me/mean/test/node_modules/grunt-node-inspector/node_modules/node-inspector/node_modules/v8-debug
gyp ERR! node -v v4.2.1
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok 
```

```
npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v4.2.1
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE

npm ERR! v8-debug@0.4.6 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the v8-debug@0.4.6 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the v8-debug package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls v8-debug
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/me/mean/test/npm-debug.log

    at ChildProcess.exithandler (child_process.js:203:12)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:818:16)
    at Socket.<anonymous> (internal/child_process.js:319:11)
    at emitOne (events.js:77:13)
    at Socket.emit (events.js:169:7)
    at Pipe._onclose (net.js:469:12)
```

**Comment by codydaig** [comment]
@brokyo What OS X version are you running?

**Comment by brokyo** [comment]
Ha I was actually editing as you commented that. 10.10.5

**Comment by patrickbrownsync** [comment]
I'm having the same issues on 10.10.5 as well

**Comment by achapela** [comment]
Same here on ubuntu 14.04.

**Comment by edrpls** [comment]
Experiencing same problem, running 10.10.5 too.

**Comment by superABD** [comment]
I ran into same issue. I'm on OS X 10.11.

**Comment by crg821** [comment]
Having the same issues here, any fix yet? OS X 10.10.5

**Comment by ilanbiala** [comment]
Are you getting exactly the same error as above?

**Comment by superABD** [comment]
@ilanbiala If you were asking me, then yes. I'm getting same error as above.

**Comment by sdoushi1** [comment]
Im having same Issue here, ubuntu.

**Comment by tmundt** [comment]
Will there be a fix for MacOS or an update for NodeJS 4.x.x soon? Why is meanjs stated "outofdate" atop the readme-File?

**Comment by ilanbiala** [comment]
@tmundt where do you see that?

**Comment by surprisetalk** [comment]
I encountered the same problem when trying to install meanjs with nodejs@4.2.2 and npm@2.14.10. Going into the package.json (in the cloned meanjs repo) and changing

> "grunt-node-inspector": "~0.2.0"

to

> "grunt-node-inspector": ">=0.2.0",
did the trick.

Credit goes to @russfrisch

**Comment by tmundt** [comment]
Thank yo @surprisetalk ! But the question remains: Why is meanjs marked as  "outdated"? Is mean.io an alternative to mean.js?

**Comment by surprisetalk** [comment]
@tmundt 
Sorry if I was unclear. I was using [Yeoman](http://yeoman.io/index.html) to install [meanjs.org](http://meanjs.org)'s MEAN stack with `yo meanjs`. I don't think [mean.io](http://mean.io) has any relation to this project.
That being said, I don't know who to blame for the errant versioning on grunt-node-inspector; it could be any of the many npm dependencies.

**Comment by ArikMaor** [comment]
also experiencing same issue

**Comment by ilanbiala** [comment]
@codydaig and I have worked to get rid of many of the bugs in 0.4.1, and today we released 0.4.2. Please try it out and if it resolves your issues, let us know!
